src/banana/data/genpdf/__init__.py

```diff
@@ -0,0 +1,200 @@
+
+    In labels it is possible to pass a list of PIDs or evolution basis combinations to keep in the generated PDF.
+    In order to project on custom combinations of PIDs, it is also possible to pass a list containing the desired
+    factors for each flavor.
+
+    The default behaviour is to generate only one member for a PDF set (the zero member) but it can be changed setting
+    to True the member flag.
+
+    The info_update argument is a dictionary and provide to the user a way to change the info file of the
+    generated PDF set. If a key of info_update matches with one key of the standard info file, the information
+    are updated, otherwise they are simply added.
+
+    Turning True the value of the install flag, it is possible to autmatically install the generated PDF to
+    the lhapdf directory. By default install is False.
+
+    Parameters
+    ----------
+        name : str
+            target name
+        labels :
+            list of flavors
+        parent_pdf_set :
+            parent PDF
+        all : bool
+            iterate on members
+        install : bool
+            install on LHAPDF path
+
+    Examples
+    --------
+        To generate a PDF with a fixed function `f(x,Q2)` for some flavors you can use the
+        following snippet:
+
+        >>> # f = lambda x,Q2 ... put the desired function here
+        >>> # mask = [list of active PIDs]
+        >>> generate_pdf(name, labels, parent_pdf_set={pid: f for pid in mask})
+
+        The |API| also provides the possibility to extract arbitrary flavor combinations:
+        using the debug PDF settings we can construct a "anti-QED-singlet" combination that
+        is usefull in debugging DIS codes since it does not couple in |LO|, but only
+        through the pure-singlet contributions (starting at |NNLO|)
+
+        >>> from eko import basis_rotation as br
+        >>> from banana.data import genpdf
+        >>> import numpy as np
+        >>> anti_qed_singlet = np.zeros_like(br.flavor_basis_pids, dtype=np.float_)
+        >>> anti_qed_singlet[br.flavor_basis_pids.index(1)] = -4
+        >>> anti_qed_singlet[br.flavor_basis_pids.index(-1)] = -4
+        >>> anti_qed_singlet[br.flavor_basis_pids.index(2)] = 1
+        >>> anti_qed_singlet[br.flavor_basis_pids.index(-2)] = 1
+        >>> genpdf.generate_pdf("anti_qed_singlet", [anti_qed_singlet])
+    """
+    xgrid = np.geomspace(1e-9, 1, 240)
+    Q2grid = np.geomspace(1.3, 1e5, 35)
+    pathlib.Path(name).mkdir(exist_ok=True)
+    # Checking label basis
+    is_evol = False
+    flavor_combinations = labels
+    if project.is_evolution_labels(labels):
+        is_evol = True
+        flavor_combinations = project.evol_to_flavor(labels)
+    elif project.is_pid_labels(labels):
+        labels = np.array(labels, dtype=np.int_)
+        flavor_combinations = project.pid_to_flavor(labels)
+
+    # labels = verify_labels(args.labels)
+    # collect blocks
```

Comment:
pylint is also complaining that this function is too complex and has too much branching, which maybe is true ... maybe we can split e.g. the collections of the blocks in a separate function?

Reply:
I don't know to which extent you mean "collections of the blocks", but for sure you could split in two functions the two `if isinstance(...)` bodies.